python/pyarrow/io.pxi

```diff
@@ -2080,6 +2098,12 @@ cdef class Codec(_Weakrefable):
 
         return pybuf if asbytes else out_buf
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
+        return (f"{name}("
+                f"name={self.name}, "
+                f"compression_level={self.compression_level})")
```

Comment:
Add a test for this?

Reply:
https://github.com/apache/arrow/pull/13921/commits/eed7303948245599d498cbd3bfa2e157617346a5